app/spiffworkflow/PropertiesPanel/SpiffWorkflowPropertiesProvider.js

```diff
@@ -1,29 +1,118 @@
-import scriptProps from './parts/ScriptProps';
-import { is } from 'bpmn-js/lib/util/ModelUtil';
-
+import scriptGroup, { SCRIPT_TYPE } from './parts/ScriptGroup';
+import { is, isAny } from 'bpmn-js/lib/util/ModelUtil';
+import { DataObjectSelect } from './parts/DataObjectSelect';
+import { ListGroup, isTextFieldEntryEdited } from '@bpmn-io/properties-panel';
+import { DataObjectArray } from './parts/DataObjectArray';
 const LOW_PRIORITY = 500;
 
-export default function SpiffWorkflowPropertiesProvider(propertiesPanel, translate) {
+export default function SpiffWorkflowPropertiesProvider(propertiesPanel, translate, moddle, commandStack, elementRegistry) {
   this.getGroups = function(element) {
     return function(groups) {
       if (is(element, 'bpmn:ScriptTask')) {
-        groups.push(createScriptGroup(element, translate));
+        groups.push(createScriptGroup(element, translate, moddle));
+      } else if (isAny(element, [ 'bpmn:Task', 'bpmn:CallActivity', 'bpmn:SubProcess' ])) {
+        groups.push(preScriptPostScriptGroup(element, translate, moddle));
+      }
+      if (is(element, 'bpmn:DataObjectReference')) {
+        groups.push(createDataObjectSelector(element, translate, moddle, commandStack));
+      }
+      if (is(element, 'bpmn:Process')) {
+        groups.push(createDataObjectEditor(element, translate, moddle, commandStack, elementRegistry));
       }
+
+
       return groups;
     };
   };
-
   propertiesPanel.registerProvider(LOW_PRIORITY, this);
 }
 
-SpiffWorkflowPropertiesProvider.$inject = [ 'propertiesPanel', 'translate' ];
+SpiffWorkflowPropertiesProvider.$inject = [ 'propertiesPanel', 'translate', 'moddle', 'commandStack', 'elementRegistry' ];
+
 
-function createScriptGroup(element, translate) {
-  const spiffGroup = {
-    id: 'spiff',
+/**
+ * Adds a group to the properties panel for the script task that allows you
+ * to set the script.
+ * @param element
+ * @param translate
+ * @returns The components to add to the properties panel. */
+function createScriptGroup(element, translate, moddle, commandStack) {
+  return {
+    id: 'spiff_script',
     label: translate('SpiffWorkflow Properties'),
-    entries: scriptProps(element)
+    entries: scriptGroup(element, moddle, SCRIPT_TYPE.bpmn, 'Script', 'Code to execute.')
+  };
+}
+
+/**
+ * Adds a section to the properties' panel for NON-Script tasks, so that
+ * you can define a pre-script and a post-script for modifying data as it comes and out.
+ * @param element
+ * @param translate
+ * @param moddle  For altering the underlying XML File.
+ * @returns The components to add to the properties panel.
+ */
+function preScriptPostScriptGroup(element, translate, moddle) {
+  return {
+    id: 'spiff_pre_post_scripts',
+    label: translate('Pre-Script and Post-Script'),
```

Comment:
I do sort of like including the words "Spiff Extensions" or "Spiff" for these extra scripts, but don't feel strongly. I love not including the word "Spiff" for the data objects in the panel, since those are pure BPMN and really ought not to be spiff-specific.

Reply:
I updated the Pre/Post script and the new WebForm script (in a later PR) to clearly say "Spiffworkflow ..." in the menu - so that is more clear.